### PR TITLE
Remove `should` dependency requirement

### PR DESCRIPTION
**Pull request by ruiquelhas**
Refs #2.

### CODE REVIEW

File: src/rules/require-should-assertion-execution.js [hunk 1]
```diff
@@ -5,15 +5,25 @@
...
+
+try {
+  should = require('should');
+
+  const Assertion = should.Assertion.prototype;
+
+  assertions = Object.keys(Assertion);
+  chains = Object.keys(Assertion).filter(key => typeof Assertion[key] !== 'function');
+} catch (error) {
+  console.log('\u001b[1mRun `npm install should` to enforce the `requireShouldAssertionExecution` rule.\u001b[m');
```
Comment (ruimarinho) [inline src/rules/require-should-assertion-execution.js:29]: Such bold 🍡

File: src/rules/require-should-assertion-execution.js [hunk 2]
```diff
@@ -5,15 +5,25 @@
...
+let should;
+
+try {
+  should = require('should');
+
+  const Assertion = should.Assertion.prototype;
+
+  assertions = Object.keys(Assertion);
+  chains = Object.keys(Assertion).filter(key => typeof Assertion[key] !== 'function');
+} catch (error) {
```
Comment (ruimarinho) [inline src/rules/require-should-assertion-execution.js:24]: This should be moved just after `should = require('should');` or the error should re-throw if code is not MODULE_NOT_FOUND.

**Comment by ruimarinho** [comment]
My preference is with this approach. /cc @nunofgs @fixe

File: src/rules/require-should-assertion-execution.js (hunk 2 above)
Comment (ruiquelhas) [inline src/rules/require-should-assertion-execution.js:24]: Makes sense.

File: src/rules/require-should-assertion-execution.js (hunk 1 above)
Comment (ruiquelhas) [inline src/rules/require-should-assertion-execution.js:29]: I can remove it, but I thought I should keep the message style of `jscs`.

**Comment by ruiquelhas** [comment]
@ruimarinho we should get this merged then and release a first version.

**Comment by fixe** [comment]
Ideally this should be in its own repository.

**Comment by ruiquelhas** [comment]
I actually like the idea of moving the custom rules into their own repos. Do you guys want to go forward with that?

**Comment by ruimarinho** [comment]
@ruiquelhas I've asked you about this before but you've mentioned that this was not feasible. What has  changed?

**Comment by ruiquelhas** [comment]
@ruimarinho then it was some communication issue. I remember I brought up the fact that there are not a lot of individual rules in the wild (maybe because they need to be encapsulated in a plugin). However, we are just moving the problem to a different domain, but in that case, I think it would make more sense to just have `should` as a dependency in that module.

**Comment by ruimarinho** [comment]
Ok, can you give the external rule a try then? Would it make more sense to move this to eslint considering that jscs is being merged in?

**Comment by ruimarinho** [comment]
Refs https://github.com/seegno/jscs-config-seegno/issues/27.

**Pull request closed without merging**